test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2934,48 +2986,290 @@ public void PackageBuilderPreserveFileLastWriteTime()
+                string after1980File4 = Path.Combine(directory.Path, "after1980File4.txt");
+                File.WriteAllText(before1980File1, string.Empty);
+                File.WriteAllText(before1980File2, string.Empty);
+                File.WriteAllText(after1980File1, string.Empty);
+                File.WriteAllText(after1980File2, string.Empty);
+                File.WriteAllText(after1980File3, string.Empty);
+                File.WriteAllText(after1980File3, string.Empty);
+                File.WriteAllText(after1980File4, string.Empty);
+                File.SetLastWriteTime(before1980File1, ZipFormatMinDate.AddSeconds(-2));
+                File.SetLastWriteTime(before1980File2, ZipFormatMinDate.AddSeconds(-1));
+                File.SetLastWriteTime(after1980File1, ZipFormatMinDate);
+                File.SetLastWriteTime(after1980File2, ZipFormatMinDate.AddSeconds(1));
+                File.SetLastWriteTime(after1980File3, ZipFormatMinDate.AddSeconds(2));
+                File.SetLastWriteTime(after1980File4, year2020Date);
+
+                builder.AddFiles(directory.Path, "**", "Content");
+
+                using (var stream = new MemoryStream())
+                {
+                    builder.Save(stream);
+                    stream.Seek(0, SeekOrigin.Begin);
+
+                    // Assert
+                    using (var archive = new ZipArchive(stream, ZipArchiveMode.Read, leaveOpen: true))
+                    {
+                        foreach (ZipArchiveEntry entry in archive.Entries)
+                        {
+                            string path = Path.Combine(directory.Path, entry.Name);
+
+                            // Please note: ZipArchive stream reader sometime changes LastWriteTime by another 1 second off than what "entry.LastWriteTime" has.
+                            // The FAT filesystem of DOS has a timestamp resolution of only two seconds; ZIP file records mimic this.
+                            // As a result, the built -in timestamp resolution of files in a ZIP archive is only two seconds, though extra fields can be used to store more precise timestamps.
+                            // That is why you see this datetime interval instead of actual == of datetimes.
+                            if (File.Exists(path))
+                            {
+                                if (path == after1980File1)
+                                {
+                                    Assert.True(entry.LastWriteTime.DateTime == ZipFormatMinDate);
+                                    numberOfDateNotCorrectedFiles++;
+                                }
+                                else if (path == after1980File2 || path == after1980File3)
+                                {
+                                    Assert.True(entry.LastWriteTime.DateTime >= ZipFormatMinDate && entry.LastWriteTime.DateTime <= ZipFormatMinDate.AddSeconds(2));
```

Comment:
I converted most other cases to Assert.Equal but still prefer to keep some as it's because here `precision` is usually 2 sided(up/down) but we want to test only 1 side depending on situation. So I feel using `Assert.Equal with precision` diluting(math is better to be hard) what we're testing and making hard to distinguish which files are actually updated and which are not in current edge case situation.